addons/account_edi/models/account_edi_document.py

```diff
@@ -82,7 +82,7 @@ def _prepare_jobs(self):
         for edi_doc in documents:
             move = edi_doc.move_id
             edi_format = edi_doc.edi_format_id
-            if move.is_invoice(include_receipts=True):
+            if move._is_edi_invoice(include_receipts=True):
                 doc_type = 'invoice'
             elif move.payment_id or move.statement_line_id:
                 doc_type = 'payment'
```

Comment:
what about a method ```py
def _get_edi_document_type(self, edi_format):
            if move.is_invoice(include_receipts=True):
                doc_type = 'invoice'
            elif move.payment_id or move.statement_line_id:
                doc_type = 'payment'
```
that you could inherit in your module

Reply:
Sooo... I liked it at first.

But then, what's your condition ? That the return value is not None ? It's just not explicit from the name that you can use the method itself as a boolean condition.
Then again the current names make something explicit (only invoices and payments are required for the EDI) which just isn't always the case (I guess EC's withholds are neither).
I think @jco-odoo's solution is cleaner and more to the point for now. Although to be fair we should also have `_is_edi_payment` I guess.

Still, I pushed a commit trying to merge @smetl's suggestion with how the document types are implemented in `account_move` but that would require further refactoring (to avoid naming invoices and payments as the only documents for EDIs). So it's _even more_ like opening a Borderlands 2 loot box :see_no_evil: .

@vbe-odoo feel free to test it out but maybe not high priority until we agree on an implementation (also wait for checks to pass I didn't test this locally).

Reply:
Hi, @odoostan thanks for the changes.
I can see that some of the tests failed (ci/runbot , ci/template, ci/upgrade_enterprise ).
Anyways I did a quick FUN test with: "Send & Print", "Print" action, Supplier portal ("Share" action) in some ou localizations that use EDI (MX, PE).
All those tests were OK, no issues were found.

Thanks!
cc' @pepetreshere @fvz-odoo @ren-odoo